Bluepill-cli/Bluepill-cli/Simulator/SimulatorRunner.m

```diff
@@ -190,6 +217,43 @@ + (void)createDeviceWithConfig:(BPConfiguration *)config andName:(NSString *)dev
                        }];
 }
 
++ (SimDevice *)findDeviceWithConfig:(BPConfiguration *)config andDeviceID:(NSUUID *)deviceID {
```

Comment:
To reduce your concern, I moved the printing of error messages to the caller when device lookup fail.
I think in this simply case of dictionary lookup, return of nil is good enough to indicate an error to the caller, no extra NSError object is needed. The other methods in SimulatorRunner.m which need a NSError pointer actually pass that to the completion block, but it doesn't apply in this sync method, so we should be OK.